File: wagerproof-ios-native/Wagerproof/Features/FeatureRequests/Sheets/SubmitFeatureRequestSheet.swift
import SwiftUI
import WagerproofDesign
import WagerproofStores

/// Modal form to submit a new feature request. Ports the half-sheet
/// `Modal` at the bottom of `wagerproof-mobile/app/(drawer)/(tabs)/feature-requests.tsx`
/// (lines 559–645).
///
/// Native primitives — replaces the RN custom modal:
/// - `Form` with two sections (Title, Description) for HIG-correct field
///   chrome (system inset background, dark-mode aware borders, free
///   focus management).
/// - `.presentationDetents([.medium, .large])` + `.presentationDragIndicator(.visible)`
///   matches the spec §6 sheet detents.
/// - `.submitLabel(.send)` / multi-line description via `axis: .vertical`.
///
/// Wiring:
/// - `FeatureRequestsStore.submit(...)` does the actual Supabase insert.
///   On success we dismiss; on failure we surface `lastError` inline.
/// - The user id + display name are passed in from the parent view since
///   the sheet doesn't need its own `AuthStore` subscription.
struct SubmitFeatureRequestSheet: View {
    @Environment(\.dismiss) private var dismiss
    @Environment(FeatureRequestsStore.self) private var store

    let userId: UUID
    let displayName: String?

    @State private var title: String = ""
    @State private var description: String = ""
    @FocusState private var focusedField: Field?

    private enum Field: Hashable { case title, description }

    /// Mirrors RN's basic required-field check (line 131). Disables Submit
    /// until both fields have non-whitespace content.
    private var canSubmit: Bool {
        !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !description.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
            && !store.isSubmitting
    }

    var body: some View {
        NavigationStack {
            Form {
                Section {
                    TextField(
                        "Brief description of your feature idea",
                        text: $title
                    )
                    .focused($focusedField, equals: .title)
                    .submitLabel(.next)
                    .onSubmit { focusedField = .description }
                    .autocorrectionDisabled(false)
                } header: {
                    Text("Title")
                }

                Section {
                    // `axis: .vertical` + `lineLimit(4...10)` gives an auto-
                    // expanding textarea that matches the RN multiline input.
                    TextField(
                        "Provide more details about your feature request…",
                        text: $description,
                        axis: .vertical
                    )
                    .focused($focusedField, equals: .description)
                    .lineLimit(4...10)
                    .submitLabel(.send)
                    .onSubmit {
                        Task { await submit() }
                    }
                } header: {
                    Text("Description")
                } footer: {
                    Text("Share your ideas to help us improve WagerProof. Our team will review pending submissions.")
                        .font(AppFont.caption)
                }

                if let lastError = store.lastError {
                    Section {
                        Label(lastError, systemImage: "exclamationmark.triangle.fill")
                            .foregroundStyle(Color.appAccentAmber)
                            .font(AppFont.caption)
                    }
                }
            }
            .scrollContentBackground(.hidden)
            .background(Color.appSurface)
            .navigationTitle("Submit Feature Request")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .topBarLeading) {
                    Button("Cancel") { dismiss() }
                        .tint(Color.appTextPrimary)
                }
                ToolbarItem(placement: .topBarTrailing) {
                    Button {
                        Task { await submit() }
                    } label: {
                        if store.isSubmitting {
                            ProgressView()
                        } else {
                            Text("Submit")
                                .fontWeight(.semibold)
                        }
                    }
                    .disabled(!canSubmit)
                    .tint(Color.appPrimary)
                }
            }
            .onAppear { focusedField = .title }
            // Hint to users that validation failed via warning haptic.
            .sensoryFeedback(.warning, trigger: store.lastError)
        }
    }

    private func submit() async {
        let ok = await store.submit(
            title: title,
            description: description,
            userId: userId,
            displayName: displayName
        )
        if ok {
            // RN shows an Alert.alert("Success", …) success toast. iOS-native
            // approach: rely on the sheet dismiss + .sensoryFeedback(.success)
            // on the parent view (driven by store.justSubmittedAt). Avoid
            // toast-on-top-of-dismissed-sheet jitter.
            dismiss()
        }
    }
}

#if DEBUG
#Preview("Empty form") {
    Text("…").sheet(isPresented: .constant(true)) {
        SubmitFeatureRequestSheet(
            userId: UUID(),
            displayName: "Preview"
        )
        .environment(FeatureRequestsStore())
        .presentationDetents([.medium, .large])
    }
}
#endif

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/featurerequests/SubmitFeatureRequestSheet.kt
package com.wagerproof.app.features.featurerequests

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Icon
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.design.tokens.AppTypography
import com.wagerproof.core.design.tokens.Spacing
import com.wagerproof.core.stores.FeatureRequestsStore
import kotlinx.coroutines.launch

/**
 * Port of iOS `SubmitFeatureRequestSheet.swift`. Title + multiline Description
 * fields, inline [FeatureRequestsStore.lastError], Cancel / Submit (spinner,
 * disabled until both fields trimmed-non-empty). Dismisses on success.
 */
@Composable
fun SubmitFeatureRequestSheet(
    store: FeatureRequestsStore,
    userId: String,
    displayName: String?,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier,
) {
    var title by remember { mutableStateOf("") }
    var description by remember { mutableStateOf("") }
    val scope = rememberCoroutineScope()

    val canSubmit = title.isNotBlank() && description.isNotBlank() && !store.isSubmitting

    fun submit() {
        scope.launch {
            val ok = store.submit(title = title, description = description, userId = userId, displayName = displayName)
            if (ok) onDismiss()
        }
    }

    Column(
        modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState())
            .padding(Spacing.lg),
        verticalArrangement = Arrangement.spacedBy(Spacing.md),
    ) {
        // Toolbar row: Cancel / title / Submit.
        Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
            TextButton(onClick = onDismiss) { Text("Cancel", color = AppColors.appTextPrimary) }
            Spacer(Modifier.weight(1f))
            Text("Submit Feature Request", style = AppTypography.headline, color = AppColors.appTextPrimary)
            Spacer(Modifier.weight(1f))
            TextButton(onClick = { submit() }, enabled = canSubmit) {
                if (store.isSubmitting) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(18.dp),
                        strokeWidth = 2.dp,
                        color = AppColors.appPrimary,
                    )
                } else {
                    Text("Submit", fontWeight = FontWeight.SemiBold, color = if (canSubmit) AppColors.appPrimary else AppColors.appTextMuted)
                }
            }
        }

        Text("Title", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextSecondary)
        OutlinedTextField(
            value = title,
            onValueChange = { title = it },
            placeholder = { Text("Brief description of your feature idea") },
            singleLine = true,
            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Next),
            modifier = Modifier.fillMaxWidth(),
            colors = fieldColors(),
        )

        Text("Description", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextSecondary)
        OutlinedTextField(
            value = description,
            onValueChange = { description = it },
            placeholder = { Text("Provide more details about your feature request…") },
            minLines = 4,
            keyboardOptions = KeyboardOptions(imeAction = ImeAction.Default),
            modifier = Modifier.fillMaxWidth().heightIn(min = 120.dp),
            colors = fieldColors(),
        )
        Text(
            "Share your ideas to help us improve WagerProof. Our team will review pending submissions.",
            style = AppTypography.caption,
            color = AppColors.appTextMuted,
        )

        store.lastError?.let { err ->
            Row(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.CenterVertically) {
                Icon(
                    imageVector = AppIcon.fromSystemName("exclamationmark.triangle.fill")!!.imageVector,
                    contentDescription = null,
                    tint = AppColors.appAccentAmber,
                    modifier = Modifier.size(14.dp),
                )
                Text(err, style = AppTypography.caption, color = AppColors.appAccentAmber)
            }
        }
    }
}

@Composable
private fun fieldColors() = OutlinedTextFieldDefaults.colors(
    focusedBorderColor = AppColors.appPrimary,
    unfocusedBorderColor = AppColors.appBorder,
    focusedTextColor = AppColors.appTextPrimary,
    unfocusedTextColor = AppColors.appTextPrimary,
    cursorColor = AppColors.appPrimary,
)
